### PR TITLE
Update contributing doc for testing in the doc site

### PR DESCRIPTION
**Pull request by jimmynotjim**
It's unclear how to test components changes in the documentation site. We've also made testing in a separate project much easier.

## Additions

- Added instructions for testing in the documentation site.

## Changes

- Updated the instructions for testing in an outside project.

## Testing

- Follow the instructions.

## Review

- @anselmbradford 
- @sebworks 
- @cfarm 
- @contolini 
- @Scotchester 

## Notes

- Hopefully this clears some things up for people. Feel free to tear it apart and tell me it's still unclear and I need to make it better. I'm definitely closest to the process at the moment.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,18 +51,39 @@ by first clicking the fork button on top of the repository and cloning your fork
...
+- run `npm run cf-link`
+
+In a site specific clone of this repo:
+
+- make sure you're on the gh-pages branch
+- run `npm run cf-link`
+- run `npm run build`
+- run `npm start`
+- open a browser at http://localhost:3000/ and navigate to the component you've changed
+```
```
Comment (anselmbradford) [inline CONTRIBUTING.md]: This puts the `#### Testing in an outside project` header below into code comment formatting, see the rendered file.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,18 +51,39 @@ by first clicking the fork button on top of the repository and cloning your fork
...
-If you're hacking on a component and want to test it in a local project, use `npm link`. For example:
+#### Testing in the documentation site
+
+If you're hacking on a component and want to test it in the documentation site follow the following steps.
+
+In this repo:
+- check out a working branch
+- run `npm run cf-link`
+
+In a site specific clone of this repo:
```
Comment (Scotchester) [inline CONTRIBUTING.md]: Suggestion to make this slightly more explicit: "In a second local clone of this repo in a different folder:"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -51,18 +51,39 @@ by first clicking the fork button on top of the repository and cloning your fork
...
+- make sure you're on the gh-pages branch
+- run `npm run cf-link`
+- run `npm run build`
+- run `npm start`
+- open a browser at http://localhost:3000/ and navigate to the component you've changed
+```
+
+#### Testing in an outside project
+
+If you're hacking on a component and want to test it in a local project, use NPM linking. For example:
```
Comment (Scotchester) [inline CONTRIBUTING.md]: Trivial: Technically, it's "npm" (no caps).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -51,18 +51,39 @@ by first clicking the fork button on top of the repository and cloning your fork
...
-npm run build
-cd src/cf-buttons
-npm link
+npm run cf-link # this makes all the components linkable
+git checkout [branch] # make your edits on a branch
+
 cd ~/Projects/owning-a-home
-npm link cf-buttons
+npm link cf-buttons # or whatever component you're working on
+gulp build # or whatever build process your project uses
```
Comment (Scotchester) [inline CONTRIBUTING.md]: Trivial: Maybe space the comments out a little further from the commands?

File: CONTRIBUTING.md (hunk 1 above)
Comment (jimmynotjim) [inline CONTRIBUTING.md]: Whoops, not sure what I was doing here, removed.

**Comment by anselmbradford** [comment]
👍

**Pull request merged**